internal/endpoints/awsrulesfn/partition.go

```diff
@@ -0,0 +1,68 @@
+package awsrulesfn
+
+import "regexp"
+
+// Partition provides the metadata describing an AWS partition.
+type Partition struct {
+	ID            string                     `json:"id"`
+	Regions       map[string]PartitionConfig `json:"regions"`
+	RegionRegex   string                     `json:"regionRegex"`
+	DefaultConfig PartitionConfig            `json:"outputs"`
+}
+
+// PartitionConfig provides the endpoint metadata for an AWS region or partition.
+type PartitionConfig struct {
+	Name               *string `json:"name"`
+	DNSSuffix          *string `json:"dnsSuffix"`
+	DualStackDNSSuffix *string `json:"dualStackDnsSuffix"`
+	SupportsFIPS       *bool   `json:"supportsFIPS"`
+	SupportsDualStack  *bool   `json:"supportsDualStack"`
+}
+
+const defaultPartition = "aws"
+
+// GetPartition returns an AWS [Partition] for the region provided. If the
+// partition cannot be determined nil will be returned.
+func GetPartition(partitions []Partition, region string) *PartitionConfig {
```

Comment:
**correctness**: The list of partitions is internal (`partitions`). This function should probably be internal and the public facing one wrap it. 

e.g.

```go
func GetPartition(region string): *PartitionConfig {
    return getPartition(partitions, region)
}

func getPartition(partitions []Partition, region string) *PartitionConfig { ... }
```